test/functional/sqlsrv/MsHelper.inc

```diff
@@ -28,22 +28,67 @@ const INSERT_PREPARE_PARAMS = 4;
-    public $encType;    // randomized or deterministic; default is deterministic
-    public $options;    // a string that is null by default (e.g. NOT NULL Identity (1,1) )
+    public      $dataType;      // a string that includes the size of the type if necessary (e.g., decimal(10,5))
+    public      $colName;       // column name
+    public      $encType;       // randomized or deterministic; default is deterministic
+    public      $options;       // a string that is null by default (e.g. NOT NULL Identity (1,1) )
 
-    public function __construct($dataType, $colName = null, $options = null, $deterministic = true)
+    protected   $encryptable;   // whether Always Encrypted supports this column
+
+    public function __construct($dataType, $colName = null, $options = null, $deterministic = true, $noEncrypt = false)
     {
         if (is_null($colName)) {
             $this->colName = getDefaultColname($dataType);
         } else {
             $this->colName = $colName;
         }
-        
+
         $this->encType = ($deterministic ? "deterministic" : "randomized");
+        if (empty($dataType)) {
+            echo "Data type can not be null or empty!\n";
+        }
         $this->dataType = $dataType;
         $this->options = $options;
+
+        // first asssumes the column is not encryptable
+        $this->encryptable = false;
+        if (!$noEncrypt) {
+            $this->checkIfUnsupported();
+        }
+    }
+
+    /**
+     * if Always Encrypted not supported for this data type set $noEncryption to false
+     * @return void
+     */
+    protected function checkIfUnsupported()
+    {
+        $this->encryptable = true;
+
+        // Always Encrypted is not supported for columns with the IDENTITY property
+        // or any column using one of the following datatypes:
+        //
+        // xml, timestamp, image, ntext, text, sql_variant, hierarchyid, geography, geometry, alias,
+        // user defined-types.
+        // https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/always-encrypted-database-engine
+
+        if (stripos($this->options, "identity") !== false) {
+            $this->encryptable = false;
+        // } elseif (stripos($this->dataType, "money") !== false) {
+            // // this is a limitation in ODBC, including money and smallmoney
+            // $this->encryptable = false;
+        } elseif (!strcasecmp($this->dataType, "xml") ||
+            !strcasecmp($this->dataType, "timestamp") ||
+            !strcasecmp($this->dataType, "image") ||
+            !strcasecmp($this->dataType, "ntext") ||
+            !strcasecmp($this->dataType, "text") ||
+            !strcasecmp($this->dataType, "sql_variant") ||
+            !strcasecmp($this->dataType, "hierarchyid") ||
+            !strcasecmp($this->dataType, "geography") ||
+            !strcasecmp($this->dataType, "geometry") ||
+            !strcasecmp($this->dataType, "alias")) {
```

Comment:
money and smallmoney are also not supported, why comment out?

Reply:
also it would look nicer to put all the unsupported types in an array and there an array_in function that you can use in php to find a match

Reply:
when we do not encrypt money or smallmoney one of the AE tests with money types fail...
that's why I changed the ColumnMeta to type an optional parameter to not to encrypt a particular column instead